### PR TITLE
feat: define generic etherscan-api provider, reuse it for blockscout api

### PR DESCRIPTION
**Pull request by lsheva**
- extracted generic etherscan-api provider, 
- reuse it for blockscout api, and in future for other fallbacks

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@lumerin/wallet-core",
-  "version": "1.0.24",
+  "version": "1.0.25",
```
Comment (alex-sandrk) [inline package.json]:
```suggestion
  "version": "1.0.26",
```

Comment (alex-sandrk) [reply package.json]: please pull latest dev

File: src/plugins/explorer/explorer.js [hunk 2]
```diff
@@ -132,6 +76,26 @@ class Explorer {
...
+   * @param {string} methodName 
+   * @param  {...any} args 
+   * @returns 
+   */
+  async invoke(methodName, ...args){
+    let lastErr
+
+    for (const api of this.apis){
+      try {
+        return await pRetry(()=>api[methodName](...args), { minTimeout: 5000, retries: 5 })
```
Comment (bohdan-titan) [inline src/plugins/explorer/explorer.js]: ()=>api add spacing `() => api `

**Review by bohdan-titan — APPROVED** [review]

File: src/plugins/explorer/explorer.js (hunk 2 above)
Comment (bohdan-titan) [inline src/plugins/explorer/explorer.js]:
```suggestion
        return await pRetry(() => api[methodName](...args), { minTimeout: 5000, retries: 5 })
```

File: src/plugins/explorer/api/etherscan-compatible-api.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    const params = {
+      module: 'account',
+      action: 'tokentx',
+      sort: 'desc',
+      contractaddress: tokenAddress,
+      startBlock: from,
+      endBlock: to,
+      address,
+    }
+    const { data } = await this.api.get('', { params })
```
Comment (bohdan-titan) [inline src/plugins/explorer/api/etherscan-compatible-api.js:27]: Seems that both methods have the same code, but different params. Should we create a private method for reuse?

**Review by bohdan-titan — APPROVED** [review]

**Pull request merged**